eZ/Publish/Core/FieldType/RichText/InternalLinkValidator.php

```diff
@@ -54,10 +90,34 @@ public function validate($scheme, $id)
                 default:
                     throw new InvalidArgumentException($scheme, "Given scheme '{$scheme}' is not supported.");
             }
+        } catch (UnauthorizedException $e) {
+            return true;
```

Comment:
Why? :)

Reply:
I assume that editor can link objects event without permissions for simplicity (otherwise we need to handle this case separately), but maybe I'm wrong here.

On the other hand, I assume that PAPI needs to load object before check permission, but maybe use `sudo` method form repository will be cleaner (for sure intentions will be more explicit).

Reply:
Hmm, I'd say that's a security breach, but we've tackled this issue recently. You shouldn't be allowed to link to objects you don't have a read permission to. But that's debatable on second thought.
POV @glye @andrerom?

Reply:
True story @alongosz

Reply:
What we tackled was displaying relations to unreadable content, right? And that was solved by not displaying it, rather than failing completely, IIRC.

If the editor can link to unreadable content it will not be visible, if the same logic is followed. This leads to unpredictable results for the editor - he will know that something is there, but not what, and not who it may be visible to. This makes little sense to me. I'd prefer if you cannot link to unreadable content.

But then, what happens if a higher editor previously has linked to something the current editor can't read? The editor should then be made aware that something is there, but can't be shown what it is. And what will happen when he publishes?

Perhaps it is best if you cannot edit content that links to unreadable content. But maybe that is a BC break?

Reply:
I have tested on 5.4, see comment in jira ticket. TL;DR: Legacy is not entirely sane on this subject, either.